### PR TITLE
feat(feature-flags): update to v3, json payload, bootstrap

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

- implements https://github.com/PostHog/posthog/issues/13627 for posthog-js
     - getFeatureFlayPayload(key: string): JsonType

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers

### CODE REVIEW

File: src/posthog-featureflags.ts [hunk 1]
```diff
@@ -20,17 +26,19 @@ export const parseFeatureFlagDecideResponse = (response: Partial<DecideResponse>
                     $enabled_feature_flags,
                 })
         } else {
-            // using the v2 api
+            // using the v2^ api
             persistence &&
                 persistence.register({
                     $active_feature_flags: Object.keys(flags || {}),
```
Comment (neilkakkar) [inline src/posthog-featureflags.ts:32]: change these to the constant too?

File: src/posthog-featureflags.ts [hunk 2]
```diff
@@ -20,17 +26,19 @@ export const parseFeatureFlagDecideResponse = (response: Partial<DecideResponse>
                     $enabled_feature_flags,
                 })
         } else {
-            // using the v2 api
+            // using the v2^ api
```
Comment (neilkakkar) [inline src/posthog-featureflags.ts:29]: No payloads in v2 right?

```suggestion
            // using the v3 api
```

Comment (neilkakkar) [reply src/posthog-featureflags.ts:29]: Ah well, combining the two, I see the intention..
```suggestion
            // using the v2+ api
```

**Review by neilkakkar — APPROVED** [review]
Code looks good, but try running locally with main repo before merge!

**Comment by EDsCODE** [comment]
Testing works. Needed to add a json parse! (which needs to be accounted for in all libs)

**Comment by EDsCODE** [comment]
Pretty sure the test is being flaky. Nothing has changed since the commits with working tests. Also commit [1a57680](https://github.com/PostHog/posthog-js/pull/511/commits/1a576802ae040fa234f5576802135a4ecbb541cb) similarly had this flaking test

**Pull request merged**